frontend/markdown-pdf/build.js

```diff
@@ -113,77 +157,668 @@ Promise.all([
-                                <html>
-                                <head>
-                                    <title>${inferredTitle}</title>
-                                    <style>${css}</style>
-                                </head>
-                                <body>${html}</body>
-                                </html>
-                            `;
-                        }),
-                    browser
-                        .newPage()
-                        .then(page =>
-                            page.emulateMedia('screen').then(() => page)
-                        ),
-                ]).then(async ([html, page]) => {
-                    await page.setContent(html);
-                    return page.pdf({
-                        format: 'A4',
-                        margin: {
-                            top: '25mm',
-                            right: '10mm',
-                            bottom: '25mm',
-                            left: '10mm',
-                        },
-                        displayHeaderFooter: true,
-                        headerTemplate: `
-                            <style>${css}</style>
-                            <header>
-                                <img src="data:image/svg+xml;base64,${logoSVG.toString(
-                                    'base64'
-                                )}" class="logo" />
-                                ${filePath.replace(/\.md$/, '')}
-                            </header>`,
-                        footerTemplate: `
-                            <style>${css}</style>
-                            <footer>
-                                <span class="pageNumber"></span> / <span class="totalPages"></span>
-                            </footer>`,
-                    });
-                }),
-            ]).then(([, pdf]) =>
-                fs.writeFile(
-                    path.join(outputDir, filePath.replace('.md', '.pdf')),
-                    pdf
-                )
+                    pdfs.map(pdf =>
+                        fs.writeFile(
+                            path.join(
+                                outputDir,
+                                filePath.replace('.md', '.pdf')
+                            ),
+                            pdf
+                        )
+                    ),
+                    renderEmptyPage(browser, 'A6'), // Generate an empty page we can use in our booklets
+                ])
             )
         )
-    ).finally(() => browser.close())
+    )
+        .then(([[, emptyPage]]) =>
+            Promise.all(
+                // Loop over all distinct markdown files
+                Array.from(languages.keys(), async manual => {
+                    // Check whether the markdown file exists in the root folder
+                    if (fs.existsSync(path.join(inputDir, manual))) {
+                        // Parse Frontmatter for config extraction
```

Comment:
Het parsen van de frontmatter (en het opnieuw inlezen van de file) gebeurd een aantal keer. Kunnen we dat één keer doen en het gecached houden?

Reply:
Ik kan het voor een deel verminderen, maar als ik op elke plek de zelfde gecachede frontmatter gaat het volgensmij vrij omslachtig worden